templates/vue/src/components/RootNodeButton.vue

```diff
@@ -75,15 +89,52 @@ export default {
       this.importTapestry(file)
     },
     importTapestry(file) {
+      this.isImporting = true
       const reader = new FileReader()
-      reader.onload = e => {
+      reader.onload = async e => {
+        let upload = JSON.parse(e.target.result)
+        if (!(upload["site-url"] == wpData.wpUrl)) {
+          await this.prepareImport(upload)
+        }
         client
-          .importTapestry(JSON.parse(e.target.result))
-          .then(() => location.reload()) // TODO: Change this so a refresh isn't required
-          .catch(err => (this.error = err))
+          .importTapestry(upload)
+          .then(() => {
+            this.isImporting = false
+            this.$bvModal.show("import-changelog")
+          }) // TODO: Change this so a refresh isn't required
+          .catch(err => {
+            this.error = err
+          })
       }
       reader.readAsText(file)
     },
+    async prepareImport(data) {
+      let wp_roles = await client.getAllRoles()
+      for (let node of data.nodes) {
+        const originalPerms = node["permissions"]
```

Comment:
Let's try to avoid accessing object properties using strings wherever possible (unless it's required, like `site-url`). I feel like this should be in the linter, but if not, I'll add a rule for it.